fixtures/fixtures.ts

```diff
@@ -25,3 +27,34 @@ export const test = base.extend<TestFixtures>({
     await use(inventoryPage);
   },
 });
+
+export const expect = baseExpect.extend({
+  toBeValidImage: async (received: Locator) => {
+    const image: ImageCheck = {
+      hasNaturalWidth: false,
+      hasValidSrc: false,
+    };
+
+    image.hasNaturalWidth = await received.evaluate(
```

Comment:
I would return as soon as this check fails, to not waste resources on the request if the check is failed anyway.
```typescript
const hasWidth = await received.evaluate(
      (node: HTMLImageElement) => {
        return node.naturalWidth > 0;
      },
    );

if(!hasWidth) {
  return {
    pass: false,
    message: () => 'Image is not displayed',
  }
}

// Then valid SRC goes here.
```

Reply:
Probably it is better to have 2 matchers - one for dimensions and one for src, but this is just something to think about, no need to change.

Reply:
Separated to two different matchers, left the old matcher commented (just in case)